C/tasks/update_alerts/update_alerts.cpp

```diff
@@ -0,0 +1,172 @@
+/*
+ * Fledge Update Alerts
+ *
+ * Copyright (c) 2024 Dianomic Systems
+ *
+ * Released under the Apache 2.0 Licence
+ *
+ * Author: Devki Nandan Ghildiyal
+ */
+
+#include <update_alerts.h>
+#include <logger.h>
+
+#include <cstdlib>
+#include <thread>
+#include <csignal>
+#include <fstream>
+
+using namespace std;
+
+volatile std::sig_atomic_t signalReceived = 0;
+
+
+static void signalHandler(int signal)
+{
+	signalReceived = signal;
+}
+
+
+/**
+ * Constructor for UpdateAlerts
+ */
+UpdateAlerts::UpdateAlerts(int argc, char** argv) : FledgeProcess(argc, argv)
+{
+	std::string paramName;
+	paramName = getName();
+	m_logger = new Logger(paramName);
+	m_logger->info("UpdateAlerts starting - parameters name :%s:", paramName.c_str() );
+	m_mgtClient = this->getManagementClient();
+
+}
+
+/**
+ * Destructor for UpdateAlerts
+ */
+UpdateAlerts::~UpdateAlerts()
+{
+}
+
+/**
+ * UpdateAlerts run method, called by the base class to start the process and do the actual work.
+ */
+void UpdateAlerts::run()
+{
+	// We handle these signals, add more if needed
+	std::signal(SIGINT,  signalHandler);
+	std::signal(SIGSTOP, signalHandler);
+	std::signal(SIGTERM, signalHandler);
+
+
+	if (!m_dryRun)
+	{
+		updateAlets();
+	}
+	processEnd();
+}
+
+/**
+ * Execute the updateAlets, create an alert for all the packages for which update is available
+ */
+void UpdateAlerts::updateAlets()
+{
+	m_logger->info("updateAlets running");
+	try
+	{
+		for (auto key: getUpgradablePackageList())
+		{
+			std::string message = "new update is available for " + key;
+			std::string urgency = "normal";
+			if (!m_mgtClient->raiseAlert(key,message,urgency))
+			{
+				m_logger->error("Failed to raise alert");
+			}
+
+		}
+
+	}
+	catch (...)
+	{
+		std::exception_ptr p = std::current_exception();
+		string errorInfo = (p ? p.__cxa_exception_type()->name() : "null");
+
+		m_logger->error("Failed to raise alert : %s",errorInfo.c_str());
```

Comment:
an alert

Reply:
Get the text of the exception rather than the type. You can do this by re-throwing the exception and catching it to get an exception anthem then call what() on that exception to get the text.